### PR TITLE
feat: add initial test runner for v3

### PR DESCRIPTION
**Pull request by vaeng**

### CODE REVIEW

File: bin/run.sh [hunk 1]
```diff
@@ -33,28 +33,23 @@ echo "${slug}: testing..."
...
-
-    # OPTIONAL: Manually add colors to the output to help scanning the output for errors
-    # If the test output does not contain colors to help identify failing (or passing)
-    # tests, it can be helpful to manually add colors to the output
-    # colorized_test_output=$(echo "${test_output}" \
-    #      | GREP_COLOR='01;31' grep --color=always -E -e '^(ERROR:.*|.*failed)$|$' \
-    #      | GREP_COLOR='01;32' grep --color=always -E -e '^.*passed$|$')
-
-    jq -n --arg output "${test_output}" '{version: 1, status: "fail", message: $output}' > ${results_file}
+    if [[ -f "user_output.md" && $(cat "user_output.md" | wc -c) -gt 0 ]]; then
```
Comment (glennj) [inline bin/run.sh:45]:
```suggestion
    if [[ -f "user_output.md" && -s "user_output.md" ]]; then
```
ref https://www.gnu.org/software/bash/manual/bash.html#Bash-Conditional-Expressions

File: bin/run.sh [hunk 2]
```diff
@@ -33,28 +33,23 @@ echo "${slug}: testing..."
...
+        test_result=$(jq "(.[] | select(.status == \"fail\")).output |=  \"$(cat user_output.md)\" " < output.json)
```
Comment (glennj) [inline bin/run.sh:46]:
```suggestion
        test_result=$(
            jq --arg uo "$(< user_output.md)" '
                (.[] | select(.status == "fail")).output |= $uo
            ' < output.json
        )
```
Simplify the quoting a bit

File: tests/user-output/expected_results.json
Comment (glennj) [inline tests/user-output/expected_results.json:1]: Do we document how students can have debug output?
* https://github.com/exercism/sqlite/blob/main/exercises/shared/.docs/tests.md
* https://github.com/exercism/sqlite/blob/main/docs/TESTS.md

**Review by glennj — APPROVED** [review]
Awesome work!

File: bin/run.sh [hunk 3]
```diff
@@ -33,28 +33,23 @@ echo "${slug}: testing..."
 
 # Run the tests for the provided implementation file and redirect stdout and
 # stderr to capture it
-test_output=$(false)
-# TODO: substitute "false" with the actual command to run the test:
-# test_output=$(command_to_run_tests 2>&1)
+cd "${solution_dir}"
```
Comment (IsaacG) [inline bin/run.sh:36]:
```suggestion
cd "${solution_dir}" || exit
```

File: bin/run.sh (hunk 1 above)
Comment (IsaacG) [reply bin/run.sh:45]: This script is sh and not bash. 
```suggestion
    if [ -f "user_output.md" ] && [ ... ]; then
```

File: bin/run.sh (hunk 2 above)
Comment (IsaacG) [reply bin/run.sh:46]: The < can be removed

File: tests/empty-file/empty-file_test.sql [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        ('divisible_by_2_not_divisible_by_4', 'ac227e82-ee82-4a09-9eb6-4f84331ffdb0', 1970, 0),
+        ('divisible_by_4_not_divisible_by_100', '4fe9b84c-8e65-489e-970b-856d60b8b78e', 1996, 1),
+        ('divisible_by_100_not_divisible_by_400', '78a7848f-9667-4192-ae53-87b30c9a02dd', 2100, 0),
+        ('divisible_by_400', '42ee56ad-d3e6-48f1-8e3f-c84078d916fc', 2000, 1),
+        ('divisible_by_200_not_divisible_by_400', 'c30331f6-f9f6-4881-ad38-8ca8c12520c1', 1800, 0);
+
+UPDATE tests
+SET status = 'pass'
+FROM (SELECT year, is_leap FROM leap) AS l
+WHERE (l.year, l.is_leap) = (tests.year, tests.result);
```
Comment (IsaacG) [inline tests/empty-file/empty-file_test.sql:35]: This is a good opportunity to get rid of single letter names

File: tests/user-output/expected_results.json
Comment (vaeng) [reply tests/user-output/expected_results.json:1]: I will add some information, thanks for the reminder.

File: bin/run.sh (hunk 1 above)
Comment (vaeng) [reply bin/run.sh]: > This script is sh and not bash.

@IsaacG should we make it bash?

Comment (IsaacG) [reply bin/run.sh]: I don't see any reason not to make it bash. At which point all `[` should be replaced with `[[` or `((`.

**Pull request merged**